src/dcp/dcp.c

```diff
@@ -346,6 +359,291 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
+        if (rc)
+            return -1;       
+
+        /* if no dkeys were returned move on */
+        if (dkey_number == 0)
+            continue;
+
+        char* ptr;
+        int   rc;
+        int   j;
+
+        /* parse out individual dkeys based on key length and numver of dkeys returned */
+        for (ptr = dkey_enum_buf, j = 0; j < dkey_number; j++) {
+
+            /* Print enumerated dkeys */
+            daos_key_t diov;
+            snprintf(dkey, dkey_kds[j].kd_key_len + 1, "%s", ptr);
+            d_iov_set(&diov, (void*)dkey, dkey_kds[j].kd_key_len);
+            ptr += dkey_kds[j].kd_key_len;
+
+            /* loop to enumerate akeys */
+            daos_anchor_t akey_anchor = {0}; 
+            while (!daos_anchor_is_eof(&akey_anchor)) {
+                d_sg_list_t     sgl;
+                d_iov_t         iov;
+                daos_key_desc_t akey_kds[ENUM_DESC_NR]       = {0};
+                uint32_t        akey_number                  = ENUM_DESC_NR;
+                char            akey_enum_buf[ENUM_DESC_BUF] = {0};
+                char 		akey[ENUM_KEY_BUF] 	     = {0};
+
+                sgl.sg_nr     = 1;
+                sgl.sg_nr_out = 0;
+                sgl.sg_iovs   = &iov;
+
+                d_iov_set(&iov, akey_enum_buf, ENUM_DESC_BUF);
+
+                /* get akeys */
+                rc = daos_obj_list_akey(*src_oh, DAOS_TX_NONE, &diov, &akey_number, akey_kds,
+                                        &sgl, &akey_anchor, NULL);
+                if (rc)
+                    return -1;       
+
+                /* if no akeys returned move on */
+                if (akey_number == 0)
+                    continue;
+
+                int j;
+                char* ptr;
+
+                /* parse out individual akeys based on key length and numver of dkeys returned */
+                for (ptr = akey_enum_buf, j = 0; j < akey_number; j++) {
+                    daos_key_t aiov;
+                    daos_iod_t iod;
+                    snprintf(akey, akey_kds[j].kd_key_len + 1, "%s", ptr);
+                    d_iov_set(&aiov, (void*)akey, akey_kds[j].kd_key_len);
+
+                    /* set iod values */
+                    iod.iod_nr   = 1;
+                    iod.iod_type = DAOS_IOD_SINGLE;
+                    iod.iod_size = DAOS_REC_ANY;
+
+                    d_iov_set(&iod.iod_name, (void*)akey, strlen(akey));
+
+                    /* Do a fetch (with NULL sgl) of single value type, and if that
+		     * returns iod_size == 0, then a single value does not exist. */
+                    rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, &diov, 1, &iod, NULL, NULL, NULL);
+
+                    /* if iod_size == 0 then this is a DAOS_IOD_ARRAY type */
+                    if ((int)iod.iod_size == 0) {
+                        rc = daos_copy_recx_array(&diov, &aiov, src_oh, dst_oh, &iod);
+                    } else {
+                        rc = daos_copy_recx_single(&diov, src_oh, dst_oh, &iod);
```

Comment:
I think the copy functions are logging errors here when `rc` is not 0,  but do we need to also check `rc` here as well?
Or maybe it makes sense to let the copy functions just return an error, without logging, and then only log an error here for the `rc`?
Thoughts?

Reply:
@daltonbohning I am just logging what function failed in these... I like it telling me which function failed. The function itself will return and log which error failed. So you get a message for each level. If that is too much logging I can do that? I don't have a strong opinion either way.

Reply:
It looks like this was updated to check the `rc` and log an error here as well. Thanks!